### PR TITLE
[O2B-1045] nav bar adjustment

### PR DESCRIPTION
**Pull request by xsalonx**
#### I have a JIRA ticket
- [x] branch and/or PR name(s) include(s) JIRA ID
- [ ] issue has "Fix version" assigned
- [x] issue "Status" is set to "In review"
- [x] PR labels are selected

Notable changes for users:
- Replace nav bar `LHC Fills` tab with `LHC` dropdown with two options:
    - `LHC Fills`
    - `LHC Periods`

Notable changes for developers:
- use only one method in model to toggle nav bar dropdowns
- change links id in https://github.com/AliceO2Group/Bookkeeping/pull/1240/files#diff-90f47b96a0752a7ea8dbe41825baf3804e1228ab28f024b599524e0d0ea2c551R47-R49, because of test failure after adding `LHC Fills` as `pageLink`

Changes made to the database:
- NA

### CODE REVIEW

File: lib/public/Model.js [hunk 1]
```diff
@@ -269,6 +269,15 @@ export default class Model extends Observable {
         this.notify();
     }
 
+    /**
+     * Toggle navbar LHC dropdown
+     * @returns {undefined}
+     */
+    toggleLHCDropdown() {
```
Comment (martinboulais) [inline lib/public/Model.js:276]: I think here you can extract a common toggle function, to which you simply pass a key

File: lib/public/components/NavBar/index.js [hunk 2]
```diff
@@ -81,7 +81,17 @@ const navBar = (model) => {
             pageTab(model, 'home', 'Home'),
             pageTab(model, 'log-overview', 'Log Entries'),
             pageTab(model, 'env-overview', 'Environments'),
-            pageTab(model, 'lhc-fill-overview', 'LHC Fills'),
+            h(`.navbar-dropdown${model.dropdownMenu === 'LHC' ? '.dropdown-open' : ''}`, [
```
Comment (martinboulais) [inline lib/public/components/NavBar/index.js:84]: Several things here:
First, I think you can extract a `subMenu` or `dropdownSubMenu` component.
Then, you can probably use `dropdown` utitlity to clean this out.
Then, `pageLink` can probably use `frontLink`/`absoluteFrontLink`

**Review by martinboulais — CHANGES_REQUESTED** [review]

File: lib/public/Model.js (hunk 1 above)
Comment (xsalonx) [reply lib/public/Model.js]: refactored

File: lib/public/components/NavBar/index.js (hunk 2 above)
Comment (xsalonx) [reply lib/public/components/NavBar/index.js]: refactored

File: lib/public/Model.js [hunk 3]
```diff
@@ -252,29 +252,13 @@ export default class Model extends Observable {
...
-     * Clears all navbar dropdowns
-     * @returns {undefined}
+     * Toggle navbar dropdown
+     * @param {string|null} key referencing dropdown to be toggled,
+     * default value is null, which causes closing all dropdowns
+     * @returns {void}
      */
-    clearDropdownMenu() {
-        this.dropdownMenu = false;
+    toggleDropdownMenu(key = null) {
```
Comment (martinboulais) [inline lib/public/Model.js:260]: No, toggle is different from clear. Please keep the clear function separated, and do not allow for null key.

File: lib/public/components/NavBar/index.js [hunk 4]
```diff
@@ -66,13 +66,32 @@ const aliFlpLinkTab = (aliFlpIndexPageUrl) => absoluteFrontLink(
 
 const btnGroupsDelimiter = h('.mh1', { style: { 'border-left': '1px solid var(--color-gray-dark)' } });
 
+/**
+ * Build navigation bar dropdown menu
+ * @param {Model} model model
+ * @param {string} key by which the model references dropdown menu
+ * @param {string|icon} dropdownButtonContent content of button which openes dropdown menu
+ * @param {Component[]} entries entries of the menu
+ * @param {object} [options.menuStyle] style for dropdown
```
Comment (martinboulais) [inline lib/public/components/NavBar/index.js:75]: it's rather attributes than style no?

File: lib/public/components/NavBar/index.js (hunk 2 above)
Comment (martinboulais) [reply lib/public/components/NavBar/index.js]: What I meant by `Then, pageLink can probably use frontLink/absoluteFrontLink` is that:
You should have `pageLink` and `absoluteLink` and avoid to always have the construction of `?page=`
I would rather pass an `onClick` callback than the full model.

File: lib/public/components/NavBar/index.js (hunk 4 above)
Comment (xsalonx) [reply lib/public/components/NavBar/index.js]: right

File: lib/public/Model.js (hunk 3 above)
Comment (xsalonx) [reply lib/public/Model.js]: clear function restored

File: lib/public/components/NavBar/index.js (hunk 2 above)
Comment (xsalonx) [reply lib/public/components/NavBar/index.js]: refactored

**Review by martinboulais — APPROVED** [review]

**Pull request merged**